CRM/Contact/BAO/Contact.php

```diff
@@ -450,7 +450,7 @@ public static function &create(&$params, $fixAddress = TRUE, $invokeHooks = TRUE
     }
 
     // process greetings CRM-4575, cache greetings
-    self::processGreetings($contact);
+    $contact = self::processGreetings($contact);
```

Comment:
The general change here -- updating both DB and in-memory data -- makes sense.

As a sort of style issue... it looks like the signature of this is `function processGreetings(object &$contact...) ==> $contact`, which seems surprising. You'd think that the signature `processGreetings(object $contact) ==> void` would be enough (since objects in PHP can be modified without any special pass-by-reference; the use of a reference and return should be extra).

However, IMHO this doesn't need to block the PR. Just noting because curious - maybe there's something else going on that I don't understand.